index.js

```diff
@@ -2,27 +2,59 @@
 
 var request = require('request');
 var async = require('async');
+var cheerio = require('cheerio');
 
 var host = 'http://www.generalfil.es/';
 var url = host + 'files-a/';
-var options = '/?filter=_Video_';
 
-
-var getFile = function(link, cb){
-  request(host + link, function(err, response, body) {
+var getFile = function (link, cb) {
+  request(host + link, function (err, response, body) {
     try {
       var finalLink = JSON.parse(body).link;
       cb(null, finalLink);
     } catch (e) {
-      cb(new Error('Error on response parsing for ' + host+link));
+      cb(new Error('Error on response parsing for ' + host + link));
     }
   });
 };
 
-var gf = function(title, cb) {
+var gf = function (title, options, cb) {
   var finalLinks = [];
+
   title = title.replace(/[^(\w)]+/gi, '-').toLowerCase();
-  request(url+title+options, function (error, response, body) {
+
+  var reqOptions = '/?';
+
+  // Set up type filter
+  // Archives | Audio | Video | Docs | Pictures | Applications | CD,DVD
+  if (options.types) {
+    var types = options.types;
+    reqOptions += '&filter=_';
+    if(types instanceof Array){
+      reqOptions += types[0];
+      for (var i = 1; i < types.length; i++) {
+        reqOptions += '_' + types[i];
+      }
+    } else {
+      reqOptions += types;
+    }
+    reqOptions += "_";
+  }
+
+  if (options.hosts) {
+    var hosts = options.hosts;
+    reqOptions += '&hosting=';
+    if(hosts instanceof Array){
+      reqOptions += hosts[0];
+      for (var i = 1; i < hosts.length; i++) {
+        reqOptions += ',' + hosts[i];
+      }
+    } else {
+      reqOptions += hosts;
+    }
+  }
+
+  request(url + title + reqOptions, function (error, response, body) {
     if (!error && response.statusCode == 200) {
       var links = body.match(/gf_ShowDownloadLink\([\w',?=\/&]+/gi);
```

Comment:
since we have now `cheerio` dependency, this can be rewritten using it, right? It will be nicer to use `cheerio` syntax than `RegExp`. What do you think?